### DISCUSSION TITLE
Rewrite rule doesn't work as expected in vercel.json config

### DISCUSSION THREAD

**Discussion by petdud** [vercel/community#6500] [Help] [answered]
### Summary

Hi,

I have this rule in my project to rewrite URL:
```
{
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "https://deepgallery.io/aligndraw/$1",
      "has": [
        {
          "type": "host",
          "value": "aligndraw.touch.gallery"
        }
      ]
    }
  ]
}
```

The intention is that when you access website/host aligndraw.touch.gallery, it should show deepgallery.io/aligndraw but instead it shows the root deepgallery.io - and I can't figure out how to fix it. Could you please help me to point how can I achieve my expected result?

### Steps to Reproduce

1. Visit aligndraw.touch.gallery

Expected: Rewrite rule will show deepgallery.io/aligndraw
Current: Rewrite rule is ignored and it shows incorrectly root - deepgallery.io

**Comment by amyegan** [comment]
I found your support case, and it looks like you already have an answer. I'll summarize here in case it helps anyone else as well.

Having a subdirectory as the destination complicates the rewrite. One of our Customer Success Engineers provided this example config as a possible solution, but noted that this may not account for all possible paths:
```json
{
  "rewrites": [

    {
      "source": "/",
      "destination": "https://deepgallery.io/aligndraw",
      "has": [ { "type": "host", "value": "aligndraw.touch.gallery" } ]
    },
    {
      "source": "/_next/:path*",
      "destination": "https://deepgallery.io/_next/:path*",
      "has": [ { "type": "host", "value": "aligndraw.touch.gallery" } ]
    },
    {
      "source": "/images/:path",
      "destination": "https://deepgallery.io/images/:path",
      "has": [ { "type": "host", "value": "aligndraw.touch.gallery" } ]
    },
    {
      "source": "/galleries/:path*",
      "destination": "https://deepgallery.io/galleries/:path*",
      "has": [ { "type": "host", "value": "aligndraw.touch.gallery" } ]
    },
    {
      "source": "/api/:path*",
      "destination": "https://deepgallery.io/api/:path*",
      "has": [ { "type": "host", "value": "aligndraw.touch.gallery" } ]
    },
    {
      "source": "/:path*",
      "destination": "https://deepgallery.io/aligndraw/:path*",
      "has": [ { "type": "host", "value": "aligndraw.touch.gallery" } ]
    }
  ]
}
```

They also noted that you're using Cloudflare which may further complicate things (see our [proxy guide](https://vercel.com/guides/can-i-use-a-proxy-on-top-of-my-vercel-deployment) for more details).

Unexpected domain routing behavior could be a result of a Cloudflare cache. You could test this by disabling Cloudflare for testing. Or you could add a new `*.vercel.app` domain to your project and replace your `has` property in the `vercel.json` with the new `*.vercel.app` domain. The second option will let you bypass Cloudflare to test the redirect without breaking your `deepgallery.io` deployment.

**Comment by petdud** [comment, marked as answer]
Thank you for the help. From email communication, this is the answer that worked well - using next.config.js:
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  async rewrites() {
    return {
      beforeFiles: [
        {
          source: '/',
          destination: 'https://deepgallery.io/aligndraw',
          has: [{ type: 'host', value: 'aligndraw.touch.gallery' }],
        },
        // other paths mentioned above
      ],
    };
  },
};

module.exports = nextConfig;

```

**Reply by amyegan** [reply]
Thanks for sharing your solution 🙂
